src/Microsoft.AspNet.OData.Shared/Query/SkipTokenQueryOption.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Query.Validators;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData.Query
+{
+    /// <summary>
+    /// This defines a $skiptoken OData query option for querying.
+    /// </summary>
+    public class SkipTokenQueryOption
+    {
+        /// <summary>
+        /// Initialize a new instance of <see cref="SkipQueryOption"/> based on the raw $skiptoken value and
+        /// an EdmModel from <see cref="ODataQueryContext"/>.
+        /// </summary>
+        /// <param name="rawValue">The raw value for $skiptoken query.</param>
+        /// <param name="context">The <see cref="ODataQueryContext"/> which contains the <see cref="IEdmModel"/> and some type information</param>
+        /// <param name="queryOptionParser">The <see cref="ODataQueryOptionParser"/> which is used to parse the query option.</param>
+        public SkipTokenQueryOption(string rawValue, ODataQueryContext context, ODataQueryOptionParser queryOptionParser)
+        {
+            if (context == null)
+            {
+                throw Error.ArgumentNull("context");
+            }
+
+            if (String.IsNullOrEmpty(rawValue))
+            {
+                throw Error.ArgumentNullOrEmpty("rawValue");
+            }
+
+            if (queryOptionParser == null)
+            {
+                throw Error.ArgumentNull("queryOptionParser");
+            }
+
+            RawValue = rawValue;
+            Validator = context.GetSkipTokenQueryValidator();
+            SkipTokenHandler = context.GetSkipTokenHandler();
+            Context = context;
+        }
+
+        /// <summary>
+        /// Gets the raw $skiptoken value.
+        /// </summary>
+        public string RawValue { get; private set; }
+
+        /// <summary>
+        /// Gets and sets the given <see cref="ODataQueryContext"/>.
+        /// </summary>
+        public ODataQueryContext Context { get; set; }
+
+        /// <summary>
+        /// Generates the nextlink value and consumes the skiptoken value.
+        /// </summary>
+        private SkipTokenHandler SkipTokenHandler { get; set; }
```

Comment:
>set [](start = 57, length = 3)

any reason to have this public setter, rather than always return the skiptokenvalidator from the ODataQueryContext?

Reply:
It is a private property. 

---
In reply to: [292960102](https://github.com/OData/WebApi/pull/1724#discussion_r292960102) [](ancestors = 292960102)